gapis/api/vulkan/api/renderpass_framebuffer.api

```diff
@@ -163,100 +242,210 @@ cmd VkResult vkCreateRenderPass(
       switch sType {
         case VK_STRUCTURE_TYPE_RENDER_PASS_INPUT_ATTACHMENT_ASPECT_CREATE_INFO: {
           ext := as!VkRenderPassInputAttachmentAspectCreateInfo*(next.Ptr)[0]
-          renderPass.InputAttachmentAspectInfo = new!InputAttachmentAspectInfo()
-          if ext.aspectReferenceCount != as!u32(0) {
-            if ext.pAspectReferences == null {
-              vkErrorNullPointer("VkInputAttachmentAspectReference(KHR)")
-            } else {
-              for j in (0 .. ext.aspectReferenceCount) {
-                // The intput attachment aspect info is only cached but not used for now, so the
-                // dependency can be wrong as a draw call may depends on just one aspect of a
-                // depth stencil image, as specified in the input attachment aspect info, while
-                // GAPID will still treat both aspects as dependees.
-                // TODO: Once the CL that enables us getting used descriptors from shader is in,
-                // we can pull input_attachment_index from the information of the used descriptors
-                // then use the input attachment aspect info to fix the dependency.
-                renderPass.InputAttachmentAspectInfo.AspectReferences[j] = ext.pAspectReferences[0]
-              }
-            }
+
+          for j in (0 .. ext.aspectReferenceCount) {
+            // The input attachment aspect info is only cached but not used for now, so the
+            // dependency can be wrong as a draw call may depends on just one aspect of a
+            // depth stencil image, as specified in the input attachment aspect info, while
+            // GAPID will still treat both aspects as dependees.
+            // TODO: Once the CL that enables us getting used descriptors from shader is in,
+            // we can pull input_attachment_index from the information of the used descriptors
+            // then use the input attachment aspect info to fix the dependency.
+            aspectReference := ext.pAspectReferences[0]
```

Comment:
shouldn't this use `[j]` rather than `[0]`. Looks to me like its processing the same one over and over again.

Reply:
Actually the original code does the same but I believe that the original line is incorrect as well. 

Original:

```
if ext.aspectReferenceCount != as!u32(0) {
            if ext.pAspectReferences == null {
              vkErrorNullPointer("VkInputAttachmentAspectReference(KHR)")
            } else {
              for j in (0 .. ext.aspectReferenceCount) {
                // The intput attachment aspect info is only cached but not used for now, so the
                // dependency can be wrong as a draw call may depends on just one aspect of a
                // depth stencil image, as specified in the input attachment aspect info, while
                // GAPID will still treat both aspects as dependees.
                // TODO: Once the CL that enables us getting used descriptors from shader is in,
                // we can pull input_attachment_index from the information of the used descriptors
                // then use the input attachment aspect info to fix the dependency.
                renderPass.InputAttachmentAspectInfo.AspectReferences[j] = ext.pAspectReferences[0]
              }
            }
          }
```

Probably what should have been:

```
renderPass.InputAttachmentAspectInfo.AspectReferences[j] = ext.pAspectReferences[j]
```

And in our case it should be:

```
aspectReference := ext.pAspectReferences[j]
```

Let me know what do you think about this.

Reply:
@pmuetschard

Reply:
Yes, please fix it. Just because it was broken, doesn't mean we should keep it broken :).

Reply:
My question was more on the line that "Are we sure that it's broken" or is this intended. 

I'll try to go over the line of the code again with detail and see if it actually does the correct thing. Probably it does not I you said but I want to be sure and not break it while thinking that we are fixing it.

Reply:
I've fixed this and I'll keep it as a separate commit without squashing, just in case.